core/store/src/db/testdb.rs

```diff
@@ -47,6 +48,22 @@ impl Database for TestDB {
         refcount::iter_with_rc_logic(col, iterator.into_iter())
     }
 
+    fn iter_range<'a>(
+        &'a self,
+        col: DBCol,
+        lower_bound: Option<&'a [u8]>,
+        upper_bound: Option<&'a [u8]>,
+    ) -> DBIterator<'a> {
+        let lower = lower_bound.map_or(Bound::Unbounded, |f| Bound::Included(f.to_vec()));
+        let upper = upper_bound.map_or(Bound::Unbounded, |f| Bound::Excluded(f.to_vec()));
+
+        let iterator = self.db.read().unwrap()[col]
+            .range((lower, upper))
+            .map(|(k, v)| Ok((k.clone().into_boxed_slice(), v.clone().into_boxed_slice())))
+            .collect::<Vec<io::Result<_>>>();
+        refcount::iter_with_rc_logic(col, iterator.into_iter())
+    }
```

Comment:
Isn't this reading everything into memory?

Suggestion: Remove the `collect()` + `into_iter()` and instead combine the iterators. I think just the `Map<...>` you collect could instead be directly accepted passed to `iter_with_rc_logic`. (Maybe rustc needs a bit of convincing, I didn't try it.)

And if you don't see this as an immediate necessity, please at least leave a TODO linking to a GH issue that someone can pick up. :)

Reply:
Oh yeah I see if we only use it for testdb that seems fine. Feel free to ignore in this case.

Optional: Consider adding a comment along the lines of "this is inefficient but we don't care because testdb is small" for general code readability.